src/NuGet.Clients/NuGet.VisualStudio.Common/OutputConsoleLogger.cs

```diff
@@ -30,6 +33,9 @@ public class OutputConsoleLogger : INuGetUILogger, IDisposable
         private EnvDTE.BuildEvents _buildEvents;
         private EnvDTE.SolutionEvents _solutionEvents;
 
+        [SuppressMessage("Build", "CA2213:'OutputConsoleLogger' contains field '_semaphore' that is of IDisposable type 'ReentrantSemaphore', but it is never disposed. Change the Dispose method on 'OutputConsoleLogger' to call Close or Dispose on this field.", Justification = "Field is disposed from async task invoked from Dispose.")]
+        private readonly ReentrantSemaphore _semaphore = ReentrantSemaphore.Create(0, NuGetUIThreadHelper.JoinableTaskFactory.Context, ReentrantSemaphore.ReentrancyMode.NotAllowed);
```

Comment:
Is this the same context as the one from AsyncPackage.JoinableTaskFactory. 

Specifically I am referring to https://github.com/microsoft/vs-threading/blob/babb00434437c805d926e3465ede9d8b169102d3/doc/cookbook_vs.md#how-do-i-make-sure-my-async-work-has-finished-before-my-vs-package-closes. 

Will VS shutdown safely?

Reply:
It will, because we are using NuGet's service JTF, either directly or through semaphore's ExecuteAsync. Slight improvement would be to also check disposal token, so we do not attempt to log if shutdown is in progress, but those operations are cheap anyway.

Reply:
I guess my concern is more about whether we are using the correct JTF. 
I'll defer to you here though. Hope some of the threading experts can chime in here if they see a problem.